econia-labs/emojicoin-dot-fun
# [ECO-1629] Test module/market init to 100% coverage

# Description

## Production code

1. Since the [coverage tool is broken for inline functions][coverage tool], take
   out all inline annotations except where:
    1. The function returns a borrowed reference.
    1. The function would require an `acquires` statement if not inline.
1. Add swap/buy trigger differentiation on swaps.
1. Track market-wise Aptos Coin balance in a `MarketView`.
1. Update initial market stats/state triggers.
1. Update tracking for market registration fees.
1. Add inline FDV helper function.
1. Fix start time/period return mismatch in periodic state tracker decoder.

## Test code

1. Add assorted decoders for test-only state to main module.
1. Add struct mocking with base values, and struct-wise assertion.
1. Add tests to assert all state (global storage, emitted events) modifications
   during package publication and market registration.
1. Ensure 100%

## Miscellaneous

1. Clarify nonce vs bump time in architecture specification.

# Testing

From in `src/move/emojicoin_dot_fun`

```sh
git ls-files | entr -c aptos move test --coverage --dev
```

Compare coverage against bytecode:

```sh
aptos move coverage bytecode --module emojicoin_dot_fun --dev
```

[coverage tool]: https://github.com/aptos-labs/aptos-core/issues/9154

# Checklist

- [x] Did you check off all checkboxes from the linked Linear task?
